server/src/main/java/org/opensearch/common/cache/ICacheKey.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache;
+
+import org.opensearch.common.cache.stats.CacheStatsDimension;
+
+import java.util.List;
+
+public class ICacheKey<K> {
+    public final K key; // K must implement equals()
+    public final List<CacheStatsDimension> dimensions;
+
+    public ICacheKey(K key, List<CacheStatsDimension> dimensions) {
```

Comment:
Should we allow dimensions to not be passed? Empty array by default?

ICacheKey(K key) {
 dimensions = new ArrayList<>();
}

Reply:
Sure, will add constructor like that.